pubsub/topics/main.go

```diff
@@ -145,6 +154,43 @@ func publish(client *pubsub.Client, topic, msg string) error {
 	return nil
 }
 
+func publishThatScales(client *pubsub.Client, topic string, n uint64) error {
+	ctx := context.Background()
+	var wg sync.WaitGroup
+	var n_err uint64
+	// [START pubsub_publish_with_error_handling_that_scales]
+	t := client.Topic(topic)
+
+	for i := 0; i < int(n); i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			Data: []byte("Message No. " + strconv.Itoa(i)),
+		})
+
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			// The Get method blocks until the result is returned and a
+			// server-generated ID is returned for the published message.
+			id, err := result.Get(ctx)
+			if err != nil {
+				// Error handling code can be here.
+				log.Output(1, fmt.Sprintf("Failed to publish: %v", err))
+				atomic.AddUint64(&n_err, 1)
+			} else {
+				fmt.Printf("Published message No. %d; msg ID: %v\n", i, id)
+			}
+		}(i)
+
+		wg.Wait()
+	}
+
+	if n_err > 0 {
+		return errors.New("some messages did not publish successfully")
```

Comment:
Since we're already counting the messages with errors, it might be nice to say something like `%d of %d messages did not publish successfully`